File: lab2.py
# ---Task 1 (Digital Root Sequence)---

n = int(input("Please enter a positive integer greater than 9: "))

while n <=9 :
    n = int(input("Please enter a positive integer greater than 9: "))


steps = 0
print(n, end="")

while n>9:

    temp =n
    sum = 0

    while temp > 0:

        sum = sum + temp%10
        temp = temp // 10 

    n= sum
    steps = steps+1
    print(" -> ", n, end="")

print()
print("Final value: ",n)
print("Total steps:", steps)




# ---Task 2 (FizzBuzz Counter)---

m = int(input("\nPlease enter a number between 10 and 100: "))

while m<10 or m>100:
    print("Invalid input. Please enter a number between 10 and 100: ")
    m= int(input())


fizz =0
buzz =0
fizzbuzz =0

for i in range(1, m+1):

    if i%7 ==0:

        print("(", i, "is skipped)", sep="")
        continue

    if i%3 ==0 and i%5 ==0:
        print("FizzBuzz")
        fizzbuzz +=1
    
    elif i%3 ==0 :
        print("Fizz")
        fizz +=1

    elif i%5 ==0:
        print("Buzz")
        buzz +=1

    else:
        print(i)


print("--- Summary ---")
print("Fizz count : ", fizz)
print("Buzz count : ", buzz)
print("FizzBuzz count: ", fizzbuzz)




# ---Task 3---

x = int( input("\nPlease enter a number between 3 and 9: "))

while x<3 or x>9:
    x = int( input("Please enter a number between 3 and 9: "))

for i in range(1, 2*x):

    y = x - abs(x-i)

    for j in range(1,y+1):
        print(j, end="")

    print()

File: lab2.cpp
#include <iostream>
using namespace std;

int main(){

    // ---Task 1 (Digital Root Sequence)---

    int value;
    cout << "Please enter a positive integer greater than 9: ";
    cin >> value;

    while (value <= 9 ) {
        cout<< "Please enter a positive integer greater than 9: ";
        cin >> value;

    }

    int stepCount = 0;
    cout << value;

    while (value>9) {
         
        int temp = value;
        int sum =0;

        while (temp>0) {
            sum = sum + temp%10;
            temp =temp/10;

        }
        
        value = sum;
        stepCount = stepCount+1;

        cout<< " -> " << value;

    }
    

    cout<< endl;
    cout<< "Final value: " << value << endl;
    cout<< "Total steps: " << stepCount << endl;
    
    
    
    // ---Task 2 (FizzBuzz Counter)---
    
    int n;
    cout <<"Please enter a number between 10 and 100: ";
    cin >> n;
    
    
    while (n<10|| n>100){
        cout << "Invalid input. Please enter a number between 10 and 100:" << endl;
        cin >> n;
    }
    
    int fizz =0, buzz=0, fizzbuzz=0;
    
    for(int i=1; i<=n; i++){
        
        
            
            if(i%7== 0){
                cout << "(" <<i<< " is skipped)" << endl;
                continue;
            }
            
            if(i%3== 0 && i%5 == 0){
                cout << "FizzBuzz" << endl;
                fizzbuzz++;
            }else if( i%3== 0){
                cout << "Fizz" << endl;
                fizz++;
            }else if( i%5== 0){
                cout << "Buzz" << endl;
                buzz++;
            }else{
                cout <<i<< endl;
                
            }
    
    }
    
    cout << "--- Summary ---" << endl;
    cout << "Fizz count :" << fizz<< endl;
    cout << "Buzz count :" << buzz<< endl;
    cout << "FizzBuzz count:" << fizzbuzz<< endl;
    
    
    
    // ---Task 3 ---
    
    int m;
    cout << "Please enter a number between 3 and 9:";
    cin >> m;
    
    while ( m<3 || m>9){
        cout << "Please enter a number between 3 and 9:";
        cin >> m;
        
    }
    
    for ( int j=1; j<2*m; j++){
        int k = m-abs(m-j);
        
        for (int t=1; t<=k; t++){
            cout<< t;
            
        }
        cout<< endl;
    }
    


    return 0;
     
}
